Bluetooth: Controller: Fix pdu_free_sem_give assertion under ZLI use

cvinayak opened the pull request:
Fix assertion due to multiple mayfly_enqueue calls used
under ZLI when pdu_free_sem_give is invoked from the LLL.

Fixes #46558.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix pdu_free_sem_give assertion under ZLI use

Fix assertion due to multiple mayfly_enqueue calls used
under ZLI when pdu_free_sem_give is invoked from the LLL.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

carlescufi commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c:
```diff
@@ -817,6 +817,33 @@ static void pdu_free_sem_give(void)
+	 * context.
+	 */
+	do {
+		err = k_sem_take(&sem_pdu_free, K_NO_WAIT);
```
Why not `k_sem_reset()` ?

cvinayak replied:
Was not aware of this! ... Let @caco3 verify the current fix, I will replace it with `k_sem_reset()` and verify again.

caco3 commented:
Thanks @cvinayak for the quick fix!
Your fix indeed fixes my issue!

cvinayak committed:
Bluetooth: Controller: Replace k_sem_take loop with k_sem_reset

Replace k_sem_take loop used for consuming the remaining
sem give counts with k_sem_reset.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
> Thanks @cvinayak for the quick fix! Your fix indeed fixes my issue!

Thank you, @caco3. Please test once more with the change to use k_sem_reset(), we can then merge it upstream as soon as CI passes and have your confirmation.

caco3 commented:
I can confirm that `4884840` also works.
